PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -28,7 +28,7 @@ LoadComponentCallback (
     AddMeasurePoint (0x4080);
   }
 
-  if (FeaturePcdGet (PcdMeasuredBootEnabled) && (GetFeatureCfg() & FEATURE_MEASURED_BOOT)) {
+  if (MEASURED_BOOT_ENABLED() && GetBootMode() != BOOT_ON_S3_RESUME) {
```

Comment:
in S3 resume OS osloader will not get executed.
From early stage2 control goes to OS resume.